common/src/main/scala/models/ShardedNotification.scala

```diff
@@ -13,6 +15,7 @@ object ShardRange {
 case class ShardedNotification(
   notification: Notification,
   range: ShardRange,
+  notificationAppReceivedTime: Option[Instant]
```

Comment:
This is optional for now so that any messages which are already in the queue (at deployment time) can still be processed; we'll be able to make this mandatory in a follow-up PR.